sycl/include/CL/sycl/half_type.hpp

```diff
@@ -92,6 +202,79 @@ class __SYCL_EXPORT half {
   uint16_t Buf;
 };
 
+// The main host half class
+class __SYCL_EXPORT half_new {
+public:
+  half_new() = default;
```

Comment:
```suggestion
  constexpr half_new() = default;
```

?

Reply:
This default constructor doesn't initialize half_new::Buf so it can not be constexpr